AmplifyPlugins/Core/AWSPluginsCore/Model/Support/QueryPredicate+GraphQL.swift

```diff
@@ -20,7 +25,7 @@ public struct GraphQLFilterConverter {
     /// Serialize the translated GraphQL query variable object to JSON string.
     public static func toJSON(_ queryPredicate: QueryPredicate,
                               options: JSONSerialization.WritingOptions = []) throws -> String {
-        let graphQLFilterData = try JSONSerialization.data(withJSONObject: queryPredicate.graphQLFilter,
+        let graphQLFilterData = try JSONSerialization.data(withJSONObject: queryPredicate.graphQLFilter(),
```

Comment:
this method is using a deprecated method. At first glance, it looks like the unit tests are using this it serialize to a JSON sting so it's easy to assert what a QueryPredicate looks like in JSON, but it's public, so maybe it's being used elsewhere like in DataStore? If not, we should still treat this the same way and deprecate and pass in `modelSchema` and pass that to `graphQLFilter(modelSchema)`